### PR TITLE
rgw: Normalizing X-Amz- headers for case in RGWHTTPArgs

### PR DESCRIPTION
**Pull request by pritha-srivastava**
in req_info, which are used to parse credentials in
query strings.

Signed-off-by: Pritha Srivastava <prsrivas@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by pritha-srivastava** [comment]
This addresses the issue where a presigned url has X-Amz- headers in all lower case. Without this fix, parsing of the X-Amz- headers in parse_v4_credentials fails.

**Comment by pritha-srivastava** [comment]
The HTTP headers for the same are all normalized to upper case in s->env.

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -5465,8 +5465,8 @@ AWSGeneralAbstractor::get_auth_data_v2(const req_state* const s) const
     if (now >= exp) {
       throw -EPERM;
     }
-    if (s->info.args.exists("X-Amz-Security-Token")) {
-      session_token = s->info.args.get("X-Amz-Security-Token");
+    if (s->info.args.exists("x-amz-security-Token")) {
+      session_token = s->info.args.get("x-amz-security-Token");
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc:5469]: Token -> token

**Comment by pritha-srivastava** [comment]
> lgtm -- let's test per Casey :)

Run scheduled: https://pulpito.ceph.com/prsrivas-2020-07-06_07:13:21-rgw-wip-rgw-query-string-distro-basic-smithi/

**Comment by pritha-srivastava** [comment]
> > lgtm -- let's test per Casey :)
> 
> Run scheduled: https://pulpito.ceph.com/prsrivas-2020-07-06_07:13:21-rgw-wip-rgw-query-string-distro-basic-smithi/

A lot of failures are related to s3 select, hence rebased with master and scheduled another run - https://pulpito.ceph.com/prsrivas-2020-07-06_15:40:31-rgw-wip-rgw-query-string-distro-basic-smithi/

**Comment by pritha-srivastava** [comment]
> > > lgtm -- let's test per Casey :)
> > 
> > 
> > Run scheduled: https://pulpito.ceph.com/prsrivas-2020-07-06_07:13:21-rgw-wip-rgw-query-string-distro-basic-smithi/
> 
> There are a lot of failures are related to s3 select, hence rebased with master and scheduled another run - https://pulpito.ceph.com/prsrivas-2020-07-06_15:40:31-rgw-wip-rgw-query-string-distro-basic-smithi/

Scheduled another one: https://pulpito.ceph.com/prsrivas-2020-07-07_15:57:33-rgw-wip-rgw-query-string-distro-basic-smithi/
All errors are there on master as well. There are some s3select failures. There is one failure in which ListObject is tried multiple times and then it times out. I have seen the same behaviour on another branch (http://qa-proxy.ceph.com/teuthology/aemerson-2020-07-08_01:58:03-rgw-wip-easter-bunny-distro-basic-smithi/5207350/teuthology.log). Also I tested all s3tests locally and I dont see any extra failures.

**Pull request merged**